qiskit/primitives/containers/estimator_pub.py

```diff
@@ -159,11 +176,15 @@ def coerce(cls, pub: EstimatorPubLike, precision: float | None = None) -> Estima
         if len(pub) > 3 and pub[3] is not None:
             precision = pub[3]
 
+        if len(pub) > 4 and (shape := pub[4]) is not None:
```

Comment:
```suggestion
        if len(pub) > 4 and pub[4] is not None:
```
where you're assigning it below

Reply:
fixed